src/context/GlobalProvider/index.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { createContext, useReducer } from 'react';
+
+import { globalReducer, globalState } from './reducer';
+import { GlobalState } from './interfaces';
+
+export interface GlobalContextInterface {
+  globalStore: GlobalState;
+  dispatch: React.Dispatch<any>;
+}
+
+export const GlobalContext = createContext<GlobalContextInterface>({
+  globalStore: { ...globalState },
+  // eslint-disable-next-line @typescript-eslint/no-empty-function, no-empty-function
+  dispatch: () => {}
+});
```

Comment:
I think it would've been easier and more readable to just write: 

```suggestion
export const GlobalContext = createContext());
```

And then from 21 to 24 you could write: 

```js
const GlobalProvider = ({ children}: Props) => <GlobalContext.Provider value={useReducer(globalReducer, globalState)}>{children}</GlobalContext.Provider>
```

Nevertheless. I don't like the idea of having a render component here 👀 this could easily be a higher order componente if you'd like or write in you `App/index.tsx` file since you are only going to use it once 🚀 

On our boostrap we have something like and it will work for every case, not only the global context 🙌 

```js
const withProvider = ({ Context, reducer, state }) => WrappedComponent =>
  function PropsProxy(props) {
    return (
      <Context.Provider value={useReducer(reducer, state)}>
        <WrappedComponent {...props} />
      </Context.Provider>
    );
  };
```

Reply:
Two things: 
1. for typescript rules can't use the createContext without initial values however readable as you suggest. 😅
2. Already knew the HOC that they suggest from the beginning, but I think this solution is not very efficient, first the use of this forces you to import the context, the reducer and the initial state in the component, which for him should be indifferent since that is why we are going to use a provider, and second in the future, a HOC that meets a real need may be required, and the combination and use of both could be more harmful than beneficial. 

What do you think?

Reply:
Ohhh I didn't know the first one! Good info. 

**About the second one:**  Yes, I totally agree with you. I don't really like much the idea of having to import 3 things all together in order to make it work. I think is okay if we leave it as you suggested. Though, in that case I would tell you to use an arrow function instead of a regular function. Looks off to me and we are not using hoisting benefits here so I think there is no need to make it a normal function. 

How about it ?